### PR TITLE
Adding cartographer

### PR DESCRIPTION
**Pull request by seanyen**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by seanyen** [comment]
@conda-forge/staged-recipes This is ready for review and merge. Thanks!

File: recipes/cartographer/build.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+cmake .. \
+      -G "Ninja" \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DCMAKE_PREFIX_PATH=$PREFIX \
+      -DCMAKE_INSTALL_PREFIX=$PREFIX \
+      -DCMAKE_INSTALL_LIBDIR=lib \
+      -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=ON \
+      -DBUILD_SHARED_LIBS=ON
+
+cmake --build . --config Release -- -j$CPU_COUNT
```
Comment (xhochy) [inline recipes/cartographer/build.sh:15]: No need for CPU_COUNT with Ninja.
```suggestion
cmake --build . --config Release
```

File: recipes/cartographer/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+source:
+  - url: https://github.com/cartographer-project/cartographer/archive/{{ version }}.tar.gz
+    sha256: 474a410bf6457eb8a0fd92ea412d7889fb013051e625d3ee25e8d65e4113fd6c
+    patches:
+      - fix-windows.patch  # [win]
+
+build:
+  number: 0
+  skip: true  # [unix or win and vc<14]
```
Comment (xhochy) [inline recipes/cartographer/meta.yaml:16]: Should be enough to skip unix for now, we don't build for `vc<14` anymore.
```suggestion
  skip: true  # [unix]
```

File: recipes/cartographer/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  run_exports:
+    - {{ pin_subpackage(name) }}
```
Comment (xhochy) [inline recipes/cartographer/meta.yaml:18]: Should this pin to a specific version, e.g. `x.x`

**Comment by seanyen** [comment]
@xhochy Thanks for the feedback. I fixed them and now it is ready for review.

**Comment by wolfv** [comment]
Wow!

**Pull request merged**